controllers/clusterresources/awssecuritygroupfirewallrule_controller.go

```diff
@@ -98,15 +98,32 @@ func (r *AWSSecurityGroupFirewallRuleReconciler) handleCreateFirewallRule(
 	l *logr.Logger,
 ) (ctrl.Result, error) {
 	if firewallRule.Status.ID == "" {
-		l.Info(
-			"Creating AWS security group firewall rule",
-			"cluster ID", firewallRule.Spec.ClusterID,
-			"type", firewallRule.Spec.Type,
-		)
-
+		var clusterID string
+		var err error
+		if firewallRule.Spec.ClusterRef != nil {
+			clusterID, err = GetClusterID(r.Client, ctx, firewallRule.Spec.ClusterRef)
+			if err != nil {
+				l.Error(err, "Cannot get CDCID",
+					"Cluster reference", firewallRule.Spec.ClusterRef,
+				)
+				return ctrl.Result{}, err
+			}
+			l.Info(
+				"Creating cluster network firewall rule from cluster reference",
```

Comment:
Creating cluster network firewall rule from the cluster reference

Fix in all similar cases

Reply:
done